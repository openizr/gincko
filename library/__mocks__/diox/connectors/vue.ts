/**
 * Copyright (c) Openizr. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

/**
 * diox/connectors/vue mock.
 */
const connect = vi.fn(() => vi.fn(() => ({
  steps: [{ id: 'start' }, { id: 'end' }],
  loading: process.env.LOADING === 'true',
})));

export default connect;
