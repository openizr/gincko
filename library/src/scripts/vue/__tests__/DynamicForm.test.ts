/**
 * Copyright (c) Openizr. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @jest-environment jsdom
 */

import DynamicForm from 'scripts/vue/DynamicForm.vue';
import { render, fireEvent, createEvent } from '@testing-library/vue';

jest.mock('scripts/core/Engine');
jest.mock('diox/connectors/vue');
jest.mock('scripts/vue/FormStep.vue');

describe('vue/DynamicForm', () => {
  const configuration = {
    id: 'test',
    root: 'start',
    steps: {
      start: {
        fields: {},
      },
    },
  };

  beforeEach(() => {
    jest.clearAllMocks();
    delete process.env.LOADING;
  });

  test('renders correctly - loading next step', async () => {
    process.env.LOADING = 'true';
    const { container } = render(DynamicForm, { props: { configuration } });
    expect(container.firstChild).toMatchSnapshot();
    const { container: newContainer } = render(DynamicForm, {
      props: { configuration },
      slots: { loader: '<div>LOADING</div>' },
    });
    expect(newContainer.firstChild).toMatchSnapshot();
  });

  test('renders correctly - with active step', async () => {
    const { container } = render(DynamicForm, { props: { configuration, activeStep: 'start' } });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('prevents native form submission', async () => {
    const { container } = render(DynamicForm, { props: { configuration } });
    const form = container.getElementsByTagName('form')[0];
    const event = createEvent.submit(form);
    event.preventDefault = jest.fn();
    await fireEvent(form, event);
    expect(event.preventDefault).toHaveBeenCalled();
  });
});
