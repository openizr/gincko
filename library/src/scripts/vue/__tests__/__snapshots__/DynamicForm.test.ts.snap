// Vitest Snapshot v1

exports[`vue/DynamicForm > renders correctly - with active step 1`] = `
<form
  class="gincko"
  id="test"
>
  <div
    class="gincko__steps"
  >
    
    <div
      id="step"
    >
      {"step":{"id":"start"},"index":0,"isActive":true,"customComponents":{}}
    </div>
    <div
      id="step"
    >
      {"step":{"id":"end"},"index":1,"isActive":false,"customComponents":{}}
    </div>
    
    <!--v-if-->
  </div>
</form>
`;
